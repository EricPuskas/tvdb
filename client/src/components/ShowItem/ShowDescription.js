import React from "react";

const ShowDescription = ({ overview, maxChars }) => {
  let description;
  if (overview && overview.length > 0) {
    description = overview.substring(0, maxChars);
    if (overview.length > maxChars) description += "...";
  }
  return (
    <div data-test="description">
      <h3>Overview:</h3>
      <p data-test="overview">{description}</p>
    </div>
  );
};

export default ShowDescription;
