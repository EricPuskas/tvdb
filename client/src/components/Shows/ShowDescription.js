import React from "react";

const ShowDescription = ({ overview, maxChars }) => {
  let description;
  if (overview && overview.length > 0) {
    description = overview.substring(0, maxChars);
    if (overview.length > maxChars) description += "...";
  }
  return (
    <>
      <h3>Overview:</h3>
      <p>{description}</p>
    </>
  );
};

export default ShowDescription;
