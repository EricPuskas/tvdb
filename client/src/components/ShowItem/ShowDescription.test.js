import React from "react";
import { shallow } from "enzyme";
import ShowDescription from "./ShowDescription";

describe("Show Description Component", () => {
  describe("Description over 350 chars", () => {
    const propsOver350 = {
      overview:
        "After a violent shipwreck, playboy billionaire Oliver Queen had disappeared and been presumed dead for five years before being discovered alive on a remote island in the Pacific. When he returns home to Starling City, his devoted mother Moira, the beloved sister Thea, and his best friend Tommy are going to welcome him home, but they have the feeling",
      maxChars: 350
    };
    const component = shallow(<ShowDescription {...propsOver350} />);
    it("Should pass snapshot test", () => {
      expect(component.html()).toMatchSnapshot();
    });

    it("Should trim down overview if it exceeds maxChars", () => {
      const desc = component
        .find(`[data-test="overview"]`)
        .childAt(0)
        .debug().length;
      expect(desc).toEqual(propsOver350.overview.length + 2);
    });
  });

  describe("Description equal or under 350 chars", () => {
    const propsUnderOrEqual350 = {
      overview:
        "After a violent shipwreck, playboy billionaire Oliver Queen had disappeared and been presumed dead for five years before being discovered alive on a remote island in the Pacific. When he returns home to Starling City, his devoted mother Moira, the beloved sister Thea, and his best friend Tommy are going to welcome him home, but they have the feelin",
      maxChars: 350
    };
    const component = shallow(<ShowDescription {...propsUnderOrEqual350} />);
    it("Should pass snapshot test", () => {
      expect(component.html()).toMatchSnapshot();
    });

    it("Should not trim down overview if it does not exceeds maxChars", () => {
      const desc = component
        .find(`[data-test="overview"]`)
        .childAt(0)
        .debug().length;
      expect(desc).toEqual(propsUnderOrEqual350.overview.length);
    });
  });
});
