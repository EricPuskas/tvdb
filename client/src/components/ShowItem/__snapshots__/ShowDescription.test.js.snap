// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Description Component Description equal or under 350 chars Should pass snapshot test 1`] = `"<div data-test=\\"description\\"><h3>Overview:</h3><p data-test=\\"overview\\">After a violent shipwreck, playboy billionaire Oliver Queen had disappeared and been presumed dead for five years before being discovered alive on a remote island in the Pacific. When he returns home to Starling City, his devoted mother Moira, the beloved sister Thea, and his best friend Tommy are going to welcome him home, but they have the feelin</p></div>"`;

exports[`Show Description Component Description over 350 chars Should pass snapshot test 1`] = `"<div data-test=\\"description\\"><h3>Overview:</h3><p data-test=\\"overview\\">After a violent shipwreck, playboy billionaire Oliver Queen had disappeared and been presumed dead for five years before being discovered alive on a remote island in the Pacific. When he returns home to Starling City, his devoted mother Moira, the beloved sister Thea, and his best friend Tommy are going to welcome him home, but they have the feelin...</p></div>"`;
